full-service/src/db/txo.rs

```diff
@@ -354,85 +351,45 @@ impl TxoModel for Txo {
-        let total_output_value: u64 = tx_proposal.outlays.iter().map(|o| o.value).sum();
-        let change_value: u64 = total_input_value - total_output_value - tx_proposal.fee();
-
-        // Determine whether this output is an outlay destination, or change.
-        let (value, confirmation, outlay_receiver, txo_type) = if let Some(outlay_index) =
-            tx_proposal
-                .outlay_index_to_tx_out_index
-                .iter()
-                .find_map(|(k, &v)| if v == output_index { Some(k) } else { None })
-        {
-            let outlay = &tx_proposal.outlays[*outlay_index];
-            (
-                outlay.value,
-                Some(*outlay_index),
-                Some(outlay.receiver.clone()),
-                TxoType::Payload,
-            )
-        } else {
-            // This is the change output. Note: there should only be one change
-            // output per transaction, based on how we construct
-            // transactions. If we change how we construct transactions,
-            // these assumptions will change, and should be reflected in the
-            // TxProposal.
-            (change_value, None, None, TxoType::Change)
-        };
-
-        // Update receiver, transaction_value, and transaction_txo_type, if outlay was
-        // found.
-        let (recipient_public_address_b58, subaddress_index) = if let Some(r) = outlay_receiver {
-            (b58_encode_public_address(&r)?, None)
-        } else {
-            // If not in an outlay, this output is change, according to how we build
-            // transactions.
-            ("".to_string(), Some(CHANGE_SUBADDRESS_INDEX as i64))
-        };
-
-        let encoded_confirmation = confirmation
-            .map(|p| mc_util_serial::encode(&tx_proposal.outlay_confirmation_numbers[p]));
-
-        // TODO: Update this to use the txo id of the output we are minting, not
-        // defaulting to 0
         let new_txo = NewTxo {
             id: &txo_id.to_string(),
-            value: value as i64,
             account_id: None,
-            token_id: 0,
-            target_key: &mc_util_serial::encode(&output.target_key),
-            public_key: &mc_util_serial::encode(&output.public_key),
-            e_fog_hint: &mc_util_serial::encode(&output.e_fog_hint),
-            txo: &mc_util_serial::encode(output),
-            subaddress_index,
-            key_image: None, // Only the recipient can calculate the KeyImage
+            value: output_txo.value as i64,
+            token_id: *output_txo.token_id as i64,
+            target_key: &mc_util_serial::encode(&output_txo.tx_out.target_key),
+            public_key: &mc_util_serial::encode(&output_txo.tx_out.public_key),
+            e_fog_hint: &mc_util_serial::encode(&output_txo.tx_out.e_fog_hint),
+            txo: &mc_util_serial::encode(&output_txo.tx_out),
+            subaddress_index: None,
+            key_image: None,
```

Comment:
I think the comment is nice to explain why a KeyImage is None for this type of txo

```suggestion
            key_image: None, // Only the recipient can calculate the KeyImage
```

Reply:
Fair point, this may have been an unintentional removal during my mad frenzy